hydromt_imod/imod.py

```diff
@@ -64,71 +66,316 @@ def __init__(
+            buffer=2,
+            variables="elevtn",
+            single_var_as_array=True,
+        )
+        # Reprojection
+        da = da.raster.reproject_like(self.grid, method="bilinear")
+        # Mask NoData
+        da = da.raster.mask_nodata()
+        # Add to maps
+        self.set_grid(da, "dem")
+
+    def setup_dis(self, nlay: int, bottom_fn=None, bottom_constant=None):
+        bottom = self._raster_or_constant(
+            bottom_constant, bottom_fn, self.grid["grid_2d"], method="bilinear"
+        )
+
+        top = self.grid["dem"]
+        thickness = top - bottom
+        layer_1d = xr.DataArray(
+            np.ones((nlay)), coords={"layer": np.arange(nlay) + 1}, dims=("layer",)
+        )
+        thickness_3d = layer_1d * (thickness / nlay)
+        self.set_grid(thickness_3d, "thickness")
+
+        bottom_layers = top - thickness_3d.cumsum(dim="layer")
+        bottom_layers = bottom_layers.transpose("layer", "y", "x")
+        self.set_grid(bottom_layers, "bottom")
 
-        # # add data to forcing attribute
-        # self.set_forcing(da_recharge, 'recharge')
+        idomain = xr.ones_like(bottom_layers, dtype=np.int32)
+        idomain = idomain.where(~np.isnan(top), other=0)
+        self.set_grid(idomain, "idomain")
+
+    def setup_initial_condition(self):
+        top = self.grid["dem"]
+        idomain = self.grid["idomain"]
+
+        starting_head = idomain * top
+        starting_head = starting_head.where(idomain == 1)
+
+        self.set_grid(starting_head, "starting_head")
+
+    def setup_well(self):
+        # - PCR-GLOBWB netcdf (grid)
+        # - User defined, in some csv format
         pass
 
+    def setup_storage(self, specific_yield: float, specific_storage: float):
+        """
+
+        Setup confined storage values with specific storages. To account for
+        specific yield, specific yield is set in the top layer and divided by
+        the top layer's thickness.
+
+        See also: https://deltares.gitlab.io/imod/imod-python/faq/modeling.html
+
+        """
+
+        dz_top = self.grid["thickness"].sel(layer=1)
+        specific_storage_top_layer = specific_yield / dz_top
+
+        specific_storage_grid = xr.where(
+            self.grid.coords["layer"] == 1, specific_storage_top_layer, specific_storage
+        )
+
+        specific_storage_grid = specific_storage_grid.where(self.grid["idomain"] == 1)
+
+        self.set_grid(specific_storage_grid, "specific_storage")
+
+    def setup_hydraulic_conductivity(
+        self, hydraulic_conductivity: float, vertical_anisotropy=1.0
+    ):
+        # Setup K-field:
+        # - GLIMPSE?
+        # - Scalar value
+
+        kh = xr.where(self.grid["idomain"], hydraulic_conductivity, np.nan)
+        k33 = kh / vertical_anisotropy
+
+        self.set_grid(kh, "k")
+        self.set_grid(k33, "k33")
+
+    def setup_overland_flow(self, resistance=1.0):
+        """
+        Parameters
+        ----------
+        resistance: float
```

Comment:
Would resistance be something like surface roughness (eg Manning parameter) or am I completely off? If it is then we could use the methods from hydromt_wflow based on land use to get another estimate than 1 :)